src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ManagedCertificateFinder.cs

```diff
@@ -305,11 +313,14 @@ protected virtual void Dispose(bool disposing)
 
         protected abstract X509Certificate2 CloneCertificate(X509Certificate2 cert);
 
-        private void FindCore(Predicate<X509Certificate2> predicate)
+        private void FindCore<TState>(TState state, Func<TState, X509Certificate2, bool> predicate)
         {
-            foreach (X509Certificate2 cert in _findFrom)
+            X509Certificate2Collection findFrom = _findFrom;
+            int count = findFrom.Count;
+            for (int i = 0; i < count; i++)
             {
-                if (predicate(cert))
+                X509Certificate2 cert = findFrom[i];
```

Comment:
"Why change the foreach to a for?"

-"To avoid the iterator allocation."

"But this type has a custom iterator..." (unspoken assumption, all iterators are struct iterators)

-"Yep, it's a class."

"Oh.  OK, then.  You know this API allocates a ton already, and this one probably doesn't matter, right?"

-"Yeah, but it was there."

"Fair enough."

Reply:
I love that I don't even need to be here to have this conversation :)